### DISCUSSION TITLE
Using DataSetAdapter.Get

### DISCUSSION THREAD

**Discussion by rasaliad** [viniciussanchez/RESTRequest4Delphi#96] [Q&A] [answered]
Hi Mr. Sanchez

I begin to use RESTRequest4Delphi and I want to now which property to setup to use GET with DataSetAdapter with the following JSon from a third party Application. I only need the "r" field to fill the dataset.  

Thanks, 

{
    "a": 200,
    "r": [
        {
            "category_id": 2,
            "id": 1,
            "product_price": 99.9,
            "product_code": "T001",
            "product_name": "Teddy Bear"
        },
        {
            "category_id": 2,
            "id": 3,
            "product_price": 100,
            "product_code": "T002",
            "product_name": "Optimus Prime"
        },
        {
            "category_id": 2,
            "id": 4,
            "product_price": 258,
            "product_code": "1234",
            "product_name": "Bob light year"
        },
        {
            "category_id": 2,
            "id": 6,
            "product_price": 325,
            "product_code": "monop",
            "product_name": "monopolio"
        }
    ],
    "s": "ok",
    "e": "null",
    "m": "Success"
}

**Comment by viniciussanchez** [comment, marked as answer]
You will have to get the JSON manually. And load into your dataset. JSON automatically loads the first level just... a, r, s, e, m. You can also create a Master Detail DataSet. Then it would work... but the simplest way would be to get the JSON manually and load only what you need.

```delphi
uses RESTRequest4Delphi, DataSet.Serialize;

var
  LResponse: IResponse;
begin
  LResponse := TRequest.New.BaseURL('').Get;
  if LResponse.StatusCode = 200 then
    DataSet.LoadFromJSON(LResponse.JSONValue.GetValue<TJSONArray>('r'));
```

**Reply by rasaliad** [reply]
Thank Mr. Sanchez

It works!!!,

I have to add .ToJSON , if not, the dataset get filled but and accesviolation occurs

I change from: 
```
// Fill the dataset but get a accesviolation error

mt1.LoadFromJSON(LResponse.JSONValue.GetValue<TJSONArray>('r'));
```

to: 
```
// Fill the dataset and no error occurs

mt1.LoadFromJSON(LResponse.JSONValue.GetValue<TJSONArray>('r').ToJSON);
```
Thanks again. And yours tools are very powerful, i have to learn more about it.

Rafael

**Reply by viniciussanchez** [reply]
Another alternative would be to pass the second parameter as false, like this:
```
mt1.LoadFromJSON(LResponse.JSONValue.GetValue<TJSONArray>('r'), False);
```

**Reply by mariolandossa** [reply]
@viniciussanchez como ficaria isso no Lazarus já que a propriedade GetValue não existe
